lib/internal/util/inspect.js

```diff
@@ -1869,18 +1870,18 @@ function tryStringify(arg) {
 }
 
 function format(...args) {
-  return formatWithOptionsInternal(undefined, ...args);
+  return formatWithOptionsInternal(undefined, args);
```

Comment:
Perhaps I should revert this change too and use a `SafeArrayIterator` instead like you previously did in `vm` (it got removed because of the conflict resolution but it didn't show any regressions in its benchmark run).

Reply:
`SafeArrayIterator` seems to have quite bad performance – at least, in the benchmarks I've run so far. I'd suggest reverting it.

Reply:
```
                                                                                                confidence improvement accuracy (*)   (**)   (***)
 util/format.js type='many-%' n=100000                                                                 ***    -41.17 %       ±3.51% ±4.71%  ±6.22%
 util/format.js type='no-replace-2' n=100000                                                           ***    -64.69 %       ±3.75% ±5.03%  ±6.64%
 util/format.js type='no-replace' n=100000                                                             ***    -47.45 %       ±2.60% ±3.47%  ±4.54%
 util/format.js type='number' n=100000                                                                 ***    -56.20 %       ±3.25% ±4.36%  ±5.73%
 util/format.js type='object-%s' n=100000                                                              ***    -14.76 %       ±1.98% ±2.63%  ±3.43%
 util/format.js type='object-to-string' n=100000                                                       ***    -39.71 %       ±2.30% ±3.07%  ±4.00%
 util/format.js type='only-objects' n=100000                                                           ***    -12.70 %       ±1.59% ±2.13%  ±2.78%
 util/format.js type='replace-object' n=100000                                                         ***    -33.92 %       ±2.86% ±3.83%  ±5.04%
 util/format.js type='string-2' n=100000                                                               ***    -49.84 %       ±4.50% ±6.00%  ±7.85%
 util/format.js type='string' n=100000                                                                 ***    -53.97 %       ±3.16% ±4.22%  ±5.52%
 util/format.js type='unknown' n=100000                                                                ***    -63.98 %       ±2.94% ±3.95%  ±5.21%
```

Quite bad performance was an understatement!

Reply:
Sorry, I didn't see your suggestion earlier, or else I would've reverted this one too. I guess this happens because of the additional object creation + the calls go through the methods in the subclass.

Reply:
@aduh95 reverted! Hope this one has an acceptable performance.